models/record.go

```diff
@@ -0,0 +1,241 @@
+package models
+
+import (
+	"fmt"
+	"log"
+	"net"
+	"strings"
+
+	"github.com/miekg/dns"
+	"github.com/miekg/dns/dnsutil"
+)
+
+// RecordConfig stores a DNS record.
+// Valid types:
+//   Official:
+//     A
+//     AAAA
+//     ANAME
+//     CAA
+//     CNAME
+//     MX
+//     NS
+//     PTR
+//     SRV
+//     TLSA
+//     TXT
+//   Pseudo-Types:
+//     ALIAs
+//     CF_REDIRECT
+//     CF_TEMP_REDIRECT
+//     FRAME
+//     IMPORT_TRANSFORM
+//     NAMESERVER
+//     NO_PURGE
+//     PAGE_RULE
+//     PURGE
+//     URL
+//     URL301
+type RecordConfig struct {
+	Type             string            `json:"type"`
+	Name             string            `json:"name"`   // The short name. See above.
+	Target           string            `json:"target"` // If a name, must end with "."
+	TTL              uint32            `json:"ttl,omitempty"`
+	Metadata         map[string]string `json:"meta,omitempty"`
+	NameFQDN         string            `json:"-"` // Must end with ".$origin". See above.
```

Comment:
If Record kept a pointer to its' parent domain, maybe GetNameFQDN could just be a pure function, so we don't even need to store fqdn seperately at all? Just the short name. Would be a larger change to break each provider, but compiler would help us out.